index.html

```diff
@@ -1,1642 +1,250 @@
-			<td>No</td>
-            <td>
-              <span>Attribute set for components to facilitate data storage and reading.</span>
-            </td>
-          </tr>		  
-		</tbody>
-	  </table>
-     <p>The 'refresh' component not support additional events except the common events.</p>	  
-	</section>	
-</section>
-  <section id="sec-basic-components">
-      <h2>Basic Components</h2>
-      <section>
-      <h3>image</h3>
-	  <p>The 'image' component is used to render and display images.</p>
-	  <p>In addition to common attributes, the 'image' component supports the following attributes:</p>
+  <section id="sec-scripting">
+    <h2>Scripting Resources</h2>
+    <p>
+      MiniApp user agents support scripting resources to define the business logic and operation of the components. Although the aim of this specification is not to recommend a specific scripting language, the document considers ECMAScript (broadly known as JavaScript) [[ECMASCRIPT]] as the scripting language by default.
+
+    </p>
+    <p>
+      As indicated in [[MINIAPP-PACKAGING]], each MiniApp component has a scripting file (with the `.js` extension). Since each component has its own scope, the functions and variables defined in the scripting file have global scope across the entire component, but they are isolated from the rest of the MiniApp components.     
+    </p>
+    <p>
+      Scripts in components support module management. Modules have an independent scope, so the elements defined in a module are private by default and invisible to other modules. The exposition of the concrete module variables or functions MUST be explicitly made through the export declarations. Only local modules (i.e., hosted in the same package) MAY be imported.
+    </p>
+    <p data-cite="HTML">
+      HTML resources (with the .html extension) MAY import local scripts, such as auxiliary libraries shared by the MiniApp components. These script documents SHOULD be hosted in the `/common` directory as defined in [[MINIAPP-PACKAGING]], so they will be available to any component in the package. As in the case of scripting modules, only local scripts (i.e., hosted in the same package) will be accessible to the components. Developers MAY include any scripting resource hosted in the same package from the HTML, using the `script` element, equivalent to [^script^] in HTML, with the [^script/src^] attribute to specify the URI of the document to import. The functions and variables of the imported script will have a limited scope across the entire component.
+    </p>
+    <pre class="example" title="Scripting modules imports"><code>// Static import
+import utils from '../common/utils.js'
+
+// dynamic import using require()
+const utils = require('/common/utils.js')
+
+// dynamically using import()
+const utils = async import('/common/utils.js')
+    </code></pre>
+
+    <pre class="example" title="Loading scripts from HTML"><code>&lt;script src="../common/utils.js"&gt;&lt;script&gt;</code></pre>

```

Comment:
As we know, MiniApp has the view layer and the logic layer, and the logic layer can't access the view layer directly, and can't access the dom of view layer.
So, even if a script can be imported from HTML, it can only execute a little logic.
It's a subset of js.  Wechat called it WXS, Baidu called it SJS, Alipay called it SJS too.

Reply:
Yes, I tried to include this in the [next section (scripting resources constraints)](https://raw.githack.com/espinr/miniapp-components/all-new-sections/index.html#sec-scripting-constraints).

Reply:
Merging the changes proposed to have a whole picture of the document. Feel free to open issues to suggest changes.